src/library_fs.js

```diff
@@ -145,6 +97,79 @@ FS.staticInit();` +
       }
     },
 
+    FSStream: class {
+      constructor() {
+        this.shared = {};
+#if USE_CLOSURE_COMPILER
+        this.node = null;
+        this.flags = 0;
+#endif
```

Comment:
With real ES6 classes closure requires that all members be initialized in the construtor.  Of you try to add new properties it generates a warning.  Quite a nice feature but it does mean we need to predeclare these things.

Reply:
> I don't see this ifdefing on closure compiler in the old code - why is it needed?
> 
> Also the old code seems to store the flags on `.shared` and this does not - is that an optimization?

The new code also stores the flags on `.shared`.  see the getter and setter for `flags` below.  I'm not trying to change anything with this PR.

Reply:
> If you try to add new properties later it generates a warning.

Hm but in this case properties are declared as getters and setters - I'd expect Closure to recognise those as "statically defined" properties and not require initialisation.

Reply:
Ah true, I tried and was able to remove that one line and it seems to work.  I'm not sure what happened there because I was adding those specifically on demand to fix warnings.